wgpu-core/src/snatch.rs

```diff
@@ -66,58 +60,67 @@ impl<T> fmt::Debug for Snatchable<T> {
 
 unsafe impl<T> Sync for Snatchable<T> {}
 
-struct LockTrace {
-    purpose: &'static str,
-    caller: &'static Location<'static>,
-    backtrace: Backtrace,
-}
+use trace::LockTrace;
+#[cfg(all(debug_assertions, feature = "std"))]
+mod trace {
```

Comment:
I think it might make sense to split these `LockTrace` changes out into another PR. It has the advantage of not creating an unused `SNATCH_LOCK_TRACE` thread local on non-`debug_assertions` builds. Let me know whether you think it is worth splitting.

Reply:
On second thought, it definitely makes sense to split them into another *commit*.